camel-idea-plugin/src/main/java/com/github/cameltooling/idea/completion/header/CamelHeaderNameCompletion.java

```diff
@@ -178,22 +180,96 @@ protected abstract LookupElementBuilder createLookupElementBuilder(PsiElement el
     /**
      * Assigns the icon that matches the best with the given header to the given builder.
      */
+    abstract class EndpointHeaderIconBuilder {
+        protected final ComponentModel.EndpointHeaderModel header;
+
+        public EndpointHeaderIconBuilder(ComponentModel.EndpointHeaderModel header) {
+            this.header = header;
+        }
+
+        public abstract @Nullable Icon getIcon();
+    }
+
+    class RequiredEndpointHeaderIconBuilder extends EndpointHeaderIconBuilder {
+        public RequiredEndpointHeaderIconBuilder(ComponentModel.EndpointHeaderModel header) {
+            super(header);
+        }
+
+        public @Nullable Icon getIcon() {
+            return AllIcons.Toolwindows.ToolWindowFavorites;
+        }
+    }
+
+    class SecretEndpointHeaderIconBuilder extends EndpointHeaderIconBuilder {
+        public SecretEndpointHeaderIconBuilder(ComponentModel.EndpointHeaderModel header) {
+            super(header);
+        }
+
+        public @Nullable Icon getIcon() {
+            return AllIcons.Nodes.SecurityRole;
+        }
+    }
+
+    class MultiValueEndpointHeaderIconBuilder extends EndpointHeaderIconBuilder {
+        public MultiValueEndpointHeaderIconBuilder(ComponentModel.EndpointHeaderModel header) {
+            super(header);
+        }
+
+        public @Nullable Icon getIcon() {
+            return AllIcons.General.ArrowRight;
+        }
+    }
+
+    class EnumEndpointHeaderIconBuilder extends EndpointHeaderIconBuilder {
+        public EnumEndpointHeaderIconBuilder(ComponentModel.EndpointHeaderModel header) {
+            super(header);
+        }
+
+        public @Nullable Icon getIcon() {
+            return AllIcons.Nodes.Enum;
+        }
+    }
+
+    class ObjectEndpointHeaderIconBuilder extends EndpointHeaderIconBuilder {
+        public ObjectEndpointHeaderIconBuilder(ComponentModel.EndpointHeaderModel header) {
+            super(header);
+        }
+
+        public @Nullable Icon getIcon() {
+            return AllIcons.Nodes.Class;
+        }
+    }
+
+    class FieldEndpointHeaderIconBuilder extends EndpointHeaderIconBuilder {
+        public FieldEndpointHeaderIconBuilder(ComponentModel.EndpointHeaderModel header) {
+            super(header);
+        }
+
+        public @Nullable Icon getIcon() {
+            return AllIcons.Nodes.Field;
+        }
+    }
+
```

Comment:
ditto, too many classes just for an icon, it is abusive, don't you agree?

Reply:
yes,Thanks for pointing